gateway-service/src/main/java/org/zowe/apiml/gateway/zaas/ZaasController.java

```diff
@@ -12,108 +12,94 @@
 import org.zowe.apiml.zaas.ZaasTokenResponse;
 
+import javax.management.ServiceNotFoundException;
+
 import static org.zowe.apiml.gateway.filters.pre.ExtractAuthSourceFilter.AUTH_SOURCE_ATTR;
 import static org.zowe.apiml.gateway.filters.pre.ExtractAuthSourceFilter.AUTH_SOURCE_PARSED_ATTR;
 import static org.zowe.apiml.security.SecurityUtils.COOKIE_AUTH_NAME;
 
 @RequiredArgsConstructor
 @RestController
 @RequestMapping(ZaasController.CONTROLLER_PATH)
-@Slf4j
 public class ZaasController {
     public static final String CONTROLLER_PATH = "gateway/zaas";
 
     private final AuthSourceService authSourceService;
-    private final MessageService messageService;
     private final PassTicketService passTicketService;
     private final ZosmfService zosmfService;
+    private final TokenCreationService tokenCreationService;
 
     @PostMapping(path = "ticket", produces = MediaType.APPLICATION_JSON_VALUE)
     @Operation(summary = "Provides PassTicket for authenticated user.")
-    public ResponseEntity<Object> getPassTicket(@RequestBody TicketRequest ticketRequest, @RequestAttribute(AUTH_SOURCE_PARSED_ATTR) AuthSource.Parsed authSourceParsed) {
-
-        if (StringUtils.isEmpty(authSourceParsed.getUserId())) {
-            return ResponseEntity
-                .status(HttpStatus.UNAUTHORIZED)
-                .build();
-        }
+    public ResponseEntity<TicketResponse> getPassTicket(@RequestBody TicketRequest ticketRequest, @RequestAttribute(AUTH_SOURCE_PARSED_ATTR) AuthSource.Parsed authSourceParsed)
+        throws IRRPassTicketGenerationException, ApplicationNameNotFoundException {
 
         final String applicationName = ticketRequest.getApplicationName();
         if (StringUtils.isBlank(applicationName)) {
-            ApiMessageView messageView = messageService.createMessage("org.zowe.apiml.security.ticket.invalidApplicationName").mapToView();
-            return ResponseEntity
-                .status(HttpStatus.BAD_REQUEST)
-                .body(messageView);
+            throw new ApplicationNameNotFoundException("ApplicationName not provided.");
         }
 
-        String ticket = null;
-        try {
-            ticket = passTicketService.generate(authSourceParsed.getUserId(), applicationName);
-        } catch (IRRPassTicketGenerationException e) {
-            ApiMessageView messageView = messageService.createMessage("org.zowe.apiml.security.ticket.generateFailed",
-                e.getErrorCode().getMessage()).mapToView();
-            return ResponseEntity
-                .status(e.getHttpStatus())
-                .body(messageView);
-        }
+        String ticket = passTicketService.generate(authSourceParsed.getUserId(), applicationName);
+
         return ResponseEntity
             .status(HttpStatus.OK)
-            .body(new TicketResponse(null, authSourceParsed.getUserId(), applicationName, ticket));
+            .body(new TicketResponse("", authSourceParsed.getUserId(), applicationName, ticket));
```

Comment:
```suggestion
            .body(new TicketResponse(null, authSourceParsed.getUserId(), applicationName, ticket));
```

An empty string or null?

Reply:
Do you prefer null over empty string? I intentionally used empty string so that the value after the serialization is clear. If null is used, then this property might be omitted or it can end up with the `"null"` value.

Reply:
Empty string is better option in my opinion.